src/Data/Nat/Primality/Factorisation.agda

```diff
@@ -163,25 +165,26 @@ private
         where open ≡-Reasoning
 
       shuffle : ∃[ bs′ ] bs ↭ a ∷ bs′
-      shuffle with ys , zs , p ← ∈-∃++ (factorisationHasAllPrimeFactors prime[a] a∣Πbs prime[bs])
-        = ys ++ zs , ↭-trans (↭-reflexive p) (shift a ys zs)
+      shuffle
+        with ys , zs , refl ← ∈-∃++ (factorisationHasAllPrimeFactors prime[a] a∣Πbs prime[bs])
+        = ys ++ zs , shift a ys zs
```

Comment:
Why are these changes necessary? If this PR is truly backwards compatible as you say, then can we remove all the changes in this file from this PR and make them in a separate one? Would be reassuring! (apart from the deprecated reasoning combinator!)

Reply:
Well that's certainly true (separate PR), and maybe it's PR mission-creep, but once you start tackling `Propositional` permutation, tackling the (very modest) downstream knock-on opportunities seemed/seems worthwhile. Can revert, once again.